### ISSUE TITLE
Separate two '/healthz' and '/predict' endpoints to use different ports for stable serving

### ISSUE THREAD

**Issue by withsmilo** [bentoml/BentoML#1768]
**Describe the bug**

When the traffic handled by the BentoML model is very high, the query to the `/healthz` endpoint sometimes fails. At this time, the BentoML Pod is restarted, so to avoid this case as much as possible, we need to separate the `/healthz` and `/predict` endpoints to use different ports for stable serving.

To implement this, we should extract the `/healthz` endpoint ([link](https://github.com/bentoml/BentoML/blob/3e5094224760fa491ac1967810ff2b49faa78ce0/bentoml/server/model_app.py#L351)) from `model_app.py` as new flask app for the production environment. Please feedback me.

**To Reproduce**

**Expected behavior**

**Screenshots/Logs**

**Environment:**
 - OS: CentOS 7
 - Python Version: 3.7.9
 - BentoML Version: 0.11.0

**Additional context**

**Comment by bojiang** [comment]
@withsmilo Good work. We are aware of this issue and plan to fix this in 1.0.
I have another idea. Now we have an async HTTP proxy(the batching layer) in front of the model server. We would make it an HTTP probe multiplexer. It would repeat the statuses of upstream model servers.

**Comment by withsmilo** [comment]
Thanks for a nice feedback, @bojiang .
I'm working on this issue now because this is urgent for our service. When I finish my works, I will make a PR with similar changes for the `bentoml-1.0` branch if you want.

**Comment by bojiang** [comment]
@withsmilo How about customize the `timeout` and `max_latency` and make them lower than the `healthz` checking interval?

**Comment by withsmilo** [comment]
@bojiang 
AFAIK, BentoML's [timeout](https://github.com/bentoml/BentoML/blob/v0.11.0/bentoml/configuration/default_bentoml.cfg#L72) means Gunicorn worker's [timeout](https://docs.gunicorn.org/en/stable/settings.html#timeout), right?

**Comment by LeoVS09** [comment]
@bojiang this not will work in case of high load, workers just will be always restarting. 
if health checks will not pass between predict requests they still will fail

**Comment by LeoVS09** [comment]
Great thanks to @withsmilo, it's exactly what I was searching for. 
I build and publish in docker hub a version of my wrapper around BentoML with your changes: [Dynamic BentoML](https://github.com/LeoVS09/dynamic-bentoml).

**Comment by withsmilo** [comment]
@LeoVS09 
I have both good and bad news. I am very happy to apply my patch to your Dynamic BentoML. But I haven't tested it yet. Does it work well?

**Comment by bojiang** [comment]
@LeoVS09 Hi. Another good news is, in the coming version, we will serve probe endpoints on async server. Thus the handling of inference requests would not block probe requests anymore.

**Comment by LeoVS09** [comment]
@withsmilo Yes, I'm tested, and it working very well. You can try `leovs09/dynamic-bentoml:0.13.1-py38-health`.
I have tested it in the development K8s cluster. Health checks work correctly. Only one problem I found, I am not sure it is caused by your changes, but the service needs much more memory on your version.

**Comment by LeoVS09** [comment]
@bojiang Great to hear it, I will be glad to update my repo on your next version, when it will be released. :)

**Comment by aarnphm** [comment]
@bojiang correct me if I'm wrong, but in the new design, users can define endpoints as long as it doesn't come across with BentoML reserved right?

**Comment by withsmilo** [comment]
@LeoVS09 
Sorry for late. I've resolved your problem(too much memory in use) via [this commit](https://github.com/withsmilo/BentoML/commit/9e980ffd9e42d226e98eb25220ffe2fa088b1ed8). You can check old and new health endpoints as like:
```python
import requests
requests.get("http://127.0.0.1:5000/healthz")
requests.get("http://127.0.0.1:8081/livez")
```

**Comment by parano** [comment]
Update: this issue is fixed properly in the upcoming version 1.0's design, where BentoML's API server have switched to ASGI from WSGI, and model inference workload will be running in a separate Runner worker, to avoid blocking the API server and access to `/health`.  In cases where heavy pre-processing code is required in the API function, we recommend users to use ASYNC to re-write the logic if possible, or moving the pre-processing code to its own Runner process via a CustomRunner https://docs.bentoml.org/en/latest/guides/runner.html
